crypto/engine/eng_rdrand.c

```diff
@@ -20,28 +20,11 @@
      defined(__x86_64) || defined(__x86_64__) || \
      defined(_M_AMD64) || defined (_M_X64)) && defined(OPENSSL_CPUID_OBJ)
 
-size_t OPENSSL_ia32_rdrand(void);
+size_t OPENSSL_ia32_rdrand_bytes(unsigned char *buf, size_t len);
 
 static int get_random_bytes(unsigned char *buf, int num)
 {
-    size_t rnd;
-
-    while (num >= (int)sizeof(size_t)) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        *((size_t *)buf) = rnd;
-        buf += sizeof(size_t);
-        num -= sizeof(size_t);
-    }
-    if (num) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        memcpy(buf, &rnd, num);
-    }
-
-    return 1;
+    return (size_t)num == OPENSSL_ia32_rdrand_bytes(buf, (size_t)num);
```

Comment:
Here is question. What is the purpose of (size_t) casts? Is it actually *just* about shutting up compiler warnings? Or would it actually be **formally** more appropriate to ensure that casts can't result in undesired outcome? I.e. check for num's positivity. Note that question is not "it never happens", but what is **formally** right thing to do prior cast.

Reply:
Good point, fixed.